Automattic/woocommerce-payments
# Add ClickTooltip and HoverTooltip components to improve mobile UX

Resolves #5851

> **Warning**
> Although this is being implemented as part of the Simplify Deposits UI project, this PR applies to all uses of the Tooltip component and changes are not hidden behind a feature flag.

#### Changes proposed in this Pull Request









This PR replaces the [Tooltip](https://github.com/Automattic/woocommerce-payments/blob/8884edeb0c2c533f9e7a3261e267d25e41cb9ef2/client/components/tooltip/index.tsx#L17) component with two new components:
  * ClickTooltip: for tooltip content that includes links or other interactive elements (documentation links). This tooltip will only display when clicked.
  * HoverTooltip: for any other tooltip content. This tooltip displays on hover events and/or click events.

**All existing instances of the `Tooltip` component have been replaced with `HoverTooltip`.**

Change #5805 will use the ClickTooltip component only for [new tooltips with links from the Overview page](https://github.com/Automattic/woocommerce-payments/pull/5862#issuecomment-1487571028).

### Testing instructions

1. Go to any page within WCPay where tooltips are used.
   - Go to **Payments → Settings → Enable new checkout experience** (first-time setup)    
      <img width="400" alt="image" src="https://user-images.githubusercontent.com/3147296/227821171-61c1a137-6733-418b-84be-1881b13675b5.png">
   - Go to **Payments → Settings → Payment Methods List** (after being enabled)  
      <img width="700" alt="image" src="https://user-images.githubusercontent.com/3147296/227820457-62834e63-eacd-4715-b0b8-5a9449f3b9ea.png">
5. Ensure the tooltip content is displayed on hover and is hidden after a 600ms mouseout delay or when hovering another tooltip.

-------------------


**Post merge**
